File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.go
func maximumHappinessSum(happiness []int, k int) int64 {
	sort.Slice(happiness, func(i, j int) bool {
		return happiness[i] > happiness[j]
	})

	var result int64
	for i := 0; i < k; i++ {
		result += int64(max(happiness[i]-i, 0))
	}

	return result
}

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.rs
use std::{cmp::*, vec};

impl Solution {
    pub fn maximum_happiness_sum(mut happiness: Vec<i32>, k: i32) -> i64 {
        let mut result: i64 = 0;

        happiness.sort_by(|a, b| b.cmp(a));
        for i in 0..k {
            result += max((happiness[i as usize] - i) as i64, 0);
        }

        result
    }
}